### ISSUE TITLE
internet.task.Clock.callLater does not fulfill IReactorTime.callLater contract

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2691]
|[<img alt="washort's avatar" src="https://avatars.githubusercontent.com/u/947487?s=50" width="50" height="50">](https://github.com/washort)| @washort reported|
|-|-|
|Trac ID|trac#2691|
|Type|defect|
|Created|2007-06-05 18:01:25Z|

`IReactorTime.callLater` is documented to return a `DelayedCall` that `.cancel()` can be called upon to cancel the delayed call. `Clock.callLater` returns the `DelayedCall` scheduled for the furthest in the future.

<details><summary>Searchable metadata</summary>

```
trac-id__2691 2691
type__defect defect
reporter__washort washort
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1181066485000000 1181066485000000
changetime__1181073377000000 1181073377000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="washort's avatar" src="https://avatars.githubusercontent.com/u/947487?s=50" width="50" height="50">](https://github.com/washort)<a name="note_2"></a>|@washort set owner to @exarkun|
|-|-|

Please review in calllater-identity-2691.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @washort|
|-|-|

test_callLaterOrdering could use one more newline between it and testAdvance.  Otherwise, looks great.

**Comment by twisted-trac** [comment]
|[<img alt="washort's avatar" src="https://avatars.githubusercontent.com/u/947487?s=50" width="50" height="50">](https://github.com/washort)<a name="note_4"></a>|@washort set status to `closed`|
|-|-|

(In [20480]) Merge calllater-identity-2691.

Makes twisted.internet.task.Clock.callLater return the DelayedCall it creates, rather than the one scheduled furthest in the future.

Author: washort
Reviewer: exarkun

Closes [#2691](https://github.com/twisted/twisted/issues/2691).

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
